src/main/java/org/kuali/kfs/kns/kim/type/DataDictionaryTypeServiceBase.java

```diff
@@ -0,0 +1,874 @@
+
+        String errorLabel = DataDictionaryTypeServiceHelper.getAttributeErrorLabel(definition);
+
+        LOG.debug("(bo, attributeName, attributeValue) = ({},{},{})",
+                objectClassName,
+                attributeName,
+                attributeValue
+        );
+
+        if (StringUtils.isNotBlank(attributeValue)) {
+            Integer maxLength = definition.getAttributeField().getMaxLength();
+            if (maxLength != null && maxLength < attributeValue.length()) {
+                errors.add(AttributeError.Builder.create(errorKey, DataDictionaryTypeServiceHelper
+                    .createErrorString(KFSKeyConstants.ERROR_MAX_LENGTH, errorLabel, maxLength.toString())).build());
+                return errors;
+            }
+            Double min = getAttributeMinValue(definition);
+            if (min != null) {
+                try {
+                    if (Double.parseDouble(attributeValue) < min) {
+                        errors.add(AttributeError.Builder.create(errorKey, DataDictionaryTypeServiceHelper
+                            .createErrorString(ERROR_INCLUSIVE_MIN, errorLabel, min.toString())).build());
+                        return errors;
+                    }
+                } catch (NumberFormatException e) {
+                    // quash; this indicates that the DD contained a min for a non-numeric attribute
+                }
+            }
+            Double max = getAttributeMaxValue(definition);
+            if (max != null) {
+                try {
+
+                    if (Double.parseDouble(attributeValue) > max) {
+                        errors.add(AttributeError.Builder.create(errorKey, DataDictionaryTypeServiceHelper
+                            .createErrorString(KFSKeyConstants.ERROR_INCLUSIVE_MAX, errorLabel, max.toString())).build());
+                        return errors;
+                    }
+                } catch (NumberFormatException e) {
+                    // quash; this indicates that the DD contained a max for a non-numeric attribute
+                }
+            }
+            //Cornell Modification: KFSPTS-27121 Start
+            //Role qualifiers have their attributes defined to be of type String. The transactional system is presuming
+            //the fromAmount and toAmount role qualifier values entered by the users for money amount comparisons have
+            //been validated by the system and only contain numerals or a period that can be converted to a KualiDecimal
+            //value during downstream processing. This check will present an error on the Person document when the user
+            //attempts to "Add" an invalid from or to amount role qualifer.
+            if (errors.isEmpty() && StringUtils.isNotBlank(attributeName) && StringUtils.isNotBlank(attributeValue)
+                    && (StringUtils.equals(attributeName, KimAttributes.FROM_AMOUNT) 
+                            || StringUtils.equals(attributeName, KimAttributes.TO_AMOUNT))) {
```

Comment:
If you want to simplify the ORed string comparison section, you could use the `StringUtils.equalsAny()` method.